pkg/drivers/powerflex.go

```diff
@@ -346,3 +346,56 @@ func ModifyPowerflexCR(yamlString string, cr csmv1.ContainerStorageModule, fileT
 	}
 	return yamlString
 }
+
+// ExtractZonesFromSecret - Reads the array config secret and returns the zone label information
+func ExtractZonesFromSecret(ctx context.Context, kube client.Client, namespace string, secret string) (map[string]string, error) {
+	log := logger.GetLogger(ctx)
+
+	arraySecret, err := utils.GetSecret(ctx, secret, namespace, kube)
+	if err != nil {
+		return nil, fmt.Errorf("reading secret [%s] error [%s]", secret, err)
+	}
+
+	type StorageArrayConfig struct {
+		SystemID string `json:"systemId"`
+		Zone     struct {
+			Label string `json:"label"`
+		} `json:"zone"`
```

Comment:
Zones in the secret will have both the 'name' and the 'label' fields, right? Do we need to keep track of both of those?

Reply:
Ahh, right, confirmed with Trevor. Was confused with a different acceptance criteria:
```
zone:
  label: <user defined label name>=<zone name>
```
Fixed. Also, fixed the AC.